central/sac/authorizer/builtin_scoped_authorizer.go

```diff
@@ -43,7 +42,7 @@ func NewBuiltInScopeChecker(ctx context.Context, roles []permissions.ResolvedRol
 	if err != nil {
 		return nil, errors.Wrap(err, "reading all clusters")
 	}
-	namespaces, err := fetchNamespaces(adminCtx)
+	namespaces, err := namespaceStore.Singleton().GetAllNamespaces(adminCtx)
```

Comment:
Wait this will still cause ballooning memory right? GetAllNamespaces will not return pointers to the objects, but clones so all the access will be fast, there will be many copies of the same object

Reply:
So far the GetAllNamespaces function was performing a store walk without clones.
I'll create a dedicated lookup function for SAC purposes. A follow-up would be to have the function returning an interface exposing only the read functions needed to compute access scopes.